taipy/core/_scheduler/_scheduler.py

```diff
@@ -102,33 +141,36 @@ def is_blocked(obj: Union[Task, Job]) -> bool:
         data_nodes = obj.task.input.values() if isinstance(obj, Job) else obj.input.values()
         return any(not _DataManager._get(dn.id).is_ready_for_reading for dn in data_nodes)
 
-    def __run(self):
-        with self.lock:
-            self.__execute_jobs()
+    @classmethod
+    def __run(cls):
+        with cls.lock:
+            cls.__execute_jobs()
 
-    def __execute_jobs(self):
-        while not self.jobs_to_run.empty() and self._dispatcher._can_execute():
-            job_to_run = self.jobs_to_run.get()
-            self._dispatcher._dispatch(job_to_run)
+    @classmethod
+    def __execute_jobs(cls):
+        while not cls.jobs_to_run.empty() and cls._dispatcher._can_execute():
+            job_to_run = cls.jobs_to_run.get()
+            cls._dispatcher._dispatch(job_to_run)
 
-    def _on_status_change(self, job: Job):
+    @classmethod
+    def _on_status_change(cls, job: Job):
         if job.is_finished():
-            if self.lock.acquire(block=False):
+            if cls.lock.acquire(block=False):
                 try:
-                    self.__unblock_jobs()
-                    self.__execute_jobs()
+                    cls.__unblock_jobs()
+                    cls.__execute_jobs()
                 except:
                     ...
                 finally:
-                    self.lock.release()
+                    cls.lock.release()
 
-    def __unblock_jobs(self):
-        jobs_to_unblock = [job for job in self.blocked_jobs if not self.is_blocked(job)]
+    @classmethod
+    def __unblock_jobs(cls):
+        jobs_to_unblock = [job for job in cls.blocked_jobs if not cls.is_blocked(job)]
```

Comment:
Could you put this _if_ in the _for_ loop directly ?
The loop is simple so we don't need to create a new list just for that.